src/boinc.cpp

```diff
@@ -0,0 +1,62 @@
+#include "boinc.h"
+#include <boost/filesystem.hpp>
+#ifdef WIN32
+#include <windows.h>
+#define KEY_WOW64_64KEY 0x0100
+#endif
+
+extern std::string GetArgument(std::string arg, std::string defaultvalue);
+
+std::string GetBoincDataDir(){
+
+    std::string path = GetArgument("boincdatadir", "");
+    if (!path.empty()){
+        return path;
+    }
+
+    #ifdef WIN32
+    HKEY hKey;
+    if (RegOpenKeyEx(HKEY_LOCAL_MACHINE,
+                     L"SOFTWARE\\Space Sciences Laboratory, U.C. Berkeley\\BOINC Setup\\",
+                     0,
+                     KEY_READ|KEY_WOW64_64KEY,
+                     &hKey) == ERROR_SUCCESS){
+
+        wchar_t szPath[MAX_PATH];
+        DWORD dwSize = sizeof(szPath);
+
+        if (RegQueryValueEx(hKey,
+                        L"DATADIR",
+                        NULL,
+                        NULL,
+                        (LPBYTE)&szPath,
+                        &dwSize) == ERROR_SUCCESS){
+            RegCloseKey(hKey);
+            std::wstring wsPath = szPath;
+            std::string path(wsPath.begin(),wsPath.end());
```

Comment:
This will fail on Asian systems but we probably want to use wstring for all file paths, so leave it for now.